pkg/clusterhealth/status.go

```diff
@@ -123,15 +123,30 @@ func PopulateLocalScannerStatus(localScannerHealthInfo *storage.ScannerHealthInf
 	desiredPods := localScannerHealthInfo.GetTotalDesiredAnalyzerPods()
 	readyPods := localScannerHealthInfo.GetTotalReadyAnalyzerPods()
 
-	if desiredPods == 0 {
+	v4DesiredPods := localScannerHealthInfo.GetTotalDesiredV4IndexerPods()
+	v4ReadyPods := localScannerHealthInfo.GetTotalReadyV4IndexerPods()
+
+	if desiredPods == 0 && v4DesiredPods == 0 {
 		return storage.ClusterHealthStatus_UNINITIALIZED
 	}
-	if localScannerHealthInfo.GetTotalReadyDbPods() == 0 {
+	if localScannerHealthInfo.GetTotalReadyDbPods() != localScannerHealthInfo.GetTotalDesiredDbPods() {
+		// If V2 database is not ready, then overall status is unhealthy.
+		return storage.ClusterHealthStatus_UNHEALTHY
+	}
+	if localScannerHealthInfo.GetTotalReadyV4DbPods() != localScannerHealthInfo.GetTotalDesiredV4DbPods() {
```

Comment:
this should only be if it's enabled, right?

Reply:
When disabled both fields will be equal (ie: both will be `0` when not set) - therefore the condition will hold true regardless if enabled or not.